pkg/render/logstorage/components/elasticsearch_metrics.go

```diff
@@ -0,0 +1,151 @@
+package component
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tigera/operator/pkg/render/common/secret"
+
+	"github.com/tigera/operator/pkg/ptr"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/render"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func ElasticsearchMetrics(
+	installation *operatorv1.InstallationSpec,
+	clusterConfig *relasticsearch.ClusterConfig,
+	esMetricsCredsSecret *corev1.Secret,
+	esCertSecret *corev1.Secret,
+	clusterDomain string,
+) render.Component {
+	return &elasticsearchMetrics{
+		installation:         installation,
+		clusterConfig:        clusterConfig,
+		esMetricsCredsSecret: esMetricsCredsSecret,
+		esCertSecret:         esCertSecret,
+		clusterDomain:        clusterDomain,
+	}
+}
+
+type elasticsearchMetrics struct {
+	installation         *operatorv1.InstallationSpec
+	esMetricsImage       string
+	esMetricsCredsSecret *corev1.Secret
+	esCertSecret         *corev1.Secret
+	clusterConfig        *relasticsearch.ClusterConfig
+	clusterDomain        string
+}
+
+func (e *elasticsearchMetrics) ResolveImages(is *operatorv1.ImageSet) error {
+	var err error
+	var errMsgs []string
+
+	reg := e.installation.Registry
+	path := e.installation.ImagePath
+
+	e.esMetricsImage, err = components.GetReference(components.ComponentElasticsearchMetrics, reg, path, is)
+	if err != nil {
+		errMsgs = append(errMsgs, err.Error())
+	}
+
+	if len(errMsgs) != 0 {
+		return fmt.Errorf(strings.Join(errMsgs, ","))
+	}
+	return nil
+}
+
+func (e *elasticsearchMetrics) Objects() (objsToCreate, objsToDelete []client.Object) {
+	var toCreate []client.Object
```

Comment:
There is nothing dynamic here, should it just be `return []client.Object{ <secret>, e.metricsService(), e.metricsDeployment()}, nil`.
No biggie if you think it should remain the way you've got it.

Reply:
I had just confused myself, the reason I did this was because you `secret.ToRuntimeObject` returns multiple secrets objects, so I usually just like to use the array expansion append uses. I can just assign toCreate though to the result of ToRunitimeObjects.